### PR TITLE
Departmental Berets

### PR DESCRIPTION
**Pull request by alsoandanswer**
dreammaker_zWU4rhTTwI

Yeah. Nothing else to say.
Also the return of the IAC beret, hooray yippee.

feedback thread : https://forums.aurorastation.org/topic/15081-departmental-berets/

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Review by fernerr — CHANGES_REQUESTED** [review]
A bunch of these are invisible when worn.

While I admire the work and effort put in here, I don't think going back to the old design of beret is desired by the playerbase at large so this'll need a feedback thread.

What these berets are worn with need to be taken into account too, as I feel some of these were made in a vacuum from their intended outfits.

**Comment by alsoandanswer** [comment]
!review

File: code/game/objects/random/random.dm [hunk 1]
```diff
@@ -1031,6 +1031,24 @@
 		/obj/item/clothing/head/softcap/captain = 1
 	)
 
+/obj/random/softcap
```
Comment (JohnWildkins) [inline code/game/objects/random/random.dm:1034]:
```suggestion
/obj/random/beret
```

You'll also need to add this to the problist in /obj/random/loot

File: code/modules/clothing/head/berets.dm [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	contained_sprite = TRUE
+	siemens_coefficient = 0.9
+
+/obj/item/clothing/head/beret/colorable
+	icon_state = "beret_colorable"
+	item_state = "beret_colorable"
+
+/obj/item/clothing/head/beret/colorable/random/Initialize()
+	. = ..()
+	color = get_random_colour(TRUE)
```
Comment (JohnWildkins) [inline code/modules/clothing/head/berets.dm:16]: This doesn't appear to set the in-hand sprite's color, jsyk.

**Review by JohnWildkins — APPROVED** [review]

**Comment by alsoandanswer** [comment]
feedback thread created https://forums.aurorastation.org/topic/15081-departmental-berets/

**Comment by alsoandanswer** [comment]
dreammaker_OnfkzvAgAs

**Comment by alsoandanswer** [comment]
its been a while now with no more feedback comments

**Review by Amunak — APPROVED** [review]
It _seems_ to be fine.

**Review by Amunak — APPROVED** [review]

File: code/modules/client/preference_setup/loadout/loadout_head.dm [hunk 3]
```diff
@@ -267,14 +278,16 @@
...
 /datum/gear/head/tcfl/New()
 	..()
 	var/tcfl = list()
-	tcfl["tcfl beret, dress"] = /obj/item/clothing/head/legion_beret
-	tcfl["tcfl beret, field"] = /obj/item/clothing/head/legion
+	tcfl["tcfl beret, dress"] = /obj/item/clothing/head/beret/legion
+	tcfl["tcfl beret, sentinel"] = /obj/item/clothing/head/beret/legion/sentinel
+	tcfl["tcfl beret, legate"] = /obj/item/clothing/head/beret/legion/legate
+	tcfl["tcfl peaked cap, legate"] = /obj/item/clothing/head/legion/legate
 	gear_tweaks += new/datum/gear_tweak/path(tcfl)
```
Comment (fernerr) [inline code/modules/client/preference_setup/loadout/loadout_head.dm:291]: Except the dress beret these should not go in the loadout, as they're something worn in the field.

**Review by fernerr — CHANGES_REQUESTED** [review]
The red tcfl field beret should be returned and added to the outfit of tcfl members.
https://github.com/Aurorastation/Aurora.3/blob/8be7a39c2568ba036b0ddbbcbd46818674ebf4b0/code/datums/outfits/ert/tcfl.dm#L4
I'm in favour of these now that the feedback in the thread has been taken into account.

**Review by fernerr — APPROVED** [review]

**Review by Amunak — APPROVED** [review]

**Pull request merged**